File: php/features.php
<?php
declare(strict_types=1);

// GameData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GameDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GameDataBaseFeature();
            case "test":
                return new GameDataTestFeature();
            default:
                return new GameDataBaseFeature();
        }
    }
}

File: py/features.py
# GameData SDK feature factory

from feature.base_feature import GameDataBaseFeature
from feature.test_feature import GameDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GameDataBaseFeature(),
        "test": lambda: GameDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
